src/Stripe.Tests.XUnit/coupons_new/coupons.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class coupon_test
+    {
+        private string previousApiBase;
+        private readonly string resourceId = "co_123";
+        private StripeCouponService service;
+
+        public coupon_test()
+        {
+            this.service = new StripeCouponService(Cache.ApiKey);
+            this.previousApiBase = StripeConfiguration.GetApiBase();
+            StripeConfiguration.SetApiBase("http://localhost:12111/v1");
+        }
+
+        ~coupon_test()
+        {
+            StripeConfiguration.SetApiBase(this.previousApiBase);
+        }
+
+        [Fact]
+        public void create_succeeds()
+        {
+            var options = new StripeCouponCreateOptions() {
+                // Add a space at the end to ensure the ID is properly URL encoded
+                // when passed in the URL for other methods
+                Id = "co_123 ",
```

Comment:
I think we were using the coupon archetype ID of `25OFF` elsewhere. It may be worth reusing that here too.

Reply:
Yeah We likely would just remove Id since it's not a recommended parameter anymore. We needed it before because it was easier/required to have a valid one for other APIs (since we needed real values). Now we can just pass co_123 for all when needed since stripe-mock does not care.